File: GXpert/GXpert.Web/Modules/Analytics/AnalyticsLog/AnalyticsLogColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Analytics.Columns;

[ColumnsScript("Analytics.AnalyticsLog")]
[BasedOnRow(typeof(AnalyticsLogRow), CheckNames = true)]
public class AnalyticsLogColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string ContentTitle { get; set; }
    public string ActivationDeviceId { get; set; }
    public string StudentPrn { get; set; }
    public DateTime StartDatetime { get; set; }
    public DateTime EndDateTime { get; set; }
    public int TimeSpent { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Analytics/AnalyticsLogColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { AnalyticsLogRow } from "./AnalyticsLogRow";

export interface AnalyticsLogColumns {
    Id: Column<AnalyticsLogRow>;
    ContentTitle: Column<AnalyticsLogRow>;
    ActivationDeviceId: Column<AnalyticsLogRow>;
    StudentPrn: Column<AnalyticsLogRow>;
    StartDatetime: Column<AnalyticsLogRow>;
    EndDateTime: Column<AnalyticsLogRow>;
    TimeSpent: Column<AnalyticsLogRow>;
}

export class AnalyticsLogColumns extends ColumnsBase<AnalyticsLogRow> {
    static readonly columnsKey = 'Analytics.AnalyticsLog';
    static readonly Fields = fieldsProxy<AnalyticsLogColumns>();
}